modules/salary/view/record.inc

```diff
@@ -176,13 +176,9 @@ elseif($_lib['input']->getProperty('action_defconf_save'))
 }
 elseif($_lib['input']->getProperty('action_salary_new'))
 {
-    list($JournalID, $tmp) = 
-        $accounting->get_next_available_journalid(
-            array('available'=>true, 'update'=>true, 'type' =>'L'));
-
-    $query="select SalaryID from salary where JournalID=$JournalID";
-    if($debug) print "$query<br>\n";
+    $query="select MAX(JournalID) as MaxJournalID from salary";
```

Comment:
No. The if on line 182 is not necessary anymore, maybe it should be removed(?)
It is not needed since when we create a new salary we select the max of all existing salaries' JournalID and when we add 1 to it we get the first free JournalID. Using that one and saving it in the field for the newly created salary will cause every next salary to get a new(free) JournalID. The old logic with selecting SalaryID was the thing that was always opening the last 0 amount(no vouchers created) salary and not allowing us to have 0 amount salaries.
@stoivo should we remove the if?